### PR TITLE
Test fixes for rawhide and trust

### PR DESCRIPTION
**Pull request by flo-renaud**
### ipatests: provide a ccache to rpcclient deletetrustdom

With samba update to samba-4.20.4, rpcclient now needs a
ccache otherwise it prompts for a password.

Fixes: https://pagure.io/freeipa/issue/9667

### test_adtrust_install: add --use-krb5-ccache to smbclient command

With samba 4.20.4 the smbclient commands needs a ccache otherwise it
prompts for a password.

Fixes: https://pagure.io/freeipa/issue/9666

### CODE REVIEW

**Comment by flo-renaud** [comment]
For future ref, links to successful runs:
- `test_adtrust_install `: [Details](http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/d79e0cda-7514-11ef-99f1-fa163efeebc0)
- `test_trust`: [Details](http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/ec587d7c-7514-11ef-881e-fa163ef83758)

Removing the temp commit

**Comment by abbra** [comment]
LGTM

File: ipatests/pytest_ipa/integration/tasks.py [hunk 1]
```diff
@@ -795,15 +795,24 @@ def remove_trust_info_from_ad(master, ad_domain, ad_hostname):
...
                   'Administrator@{}'.format(ad_domain.upper()),
                   master.config.ad_admin_password)
+    # Find cache for the user
+    cache_args = []
+    result = master.run_command(["klist"]).stdout_text
+    pattern = re.compile(r'Ticket cache: (?P<cache>.*)\n')
+    res = pattern.search(result)
+    if res:
+        cache_args = ["--use-krb5-ccache", res['cache']]
+
```
Comment (rcritten) [inline ipatests/pytest_ipa/integration/tasks.py:805]: Can you break out "Find cache for the user" into a separate task and use it in both places?

**Comment by flo-renaud** [comment]
For future ref, link to successful runs:
- `test_adtrust_install`: [Details](http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/15c7fdbe-7726-11ef-892f-fa163ed57a39)
- `test_trust`: [Details](http://freeipa-org-pr-ci.s3-website.eu-central-1.amazonaws.com/jobs/f95d3234-7725-11ef-bd4d-fa163e2a1f74)

Removing the temp commit.

**Comment by flo-renaud** [comment]
master:

* 3203afcc11487730aceb222a54cbdbaaaf371d15 ipatests: provide a ccache to rpcclient deletetrustdom
* c33e92d8954dd1578c89693e10d59d2bd4f31940 test_adtrust_install: add --use-krb5-ccache to smbclient command

**Pull request closed without merging**